Update CONTRIBUTING.md and split by language (#2386)
containers/common: CONTRIBUTING.md

- [storage](https://github.com/containers/storage)
- [libhvee](https://github.com/containers/libhvee)
- [psgo](https://github.com/containers/psgo)
- [netavark](https://github.com/containers/netavark)
- [aardvark](https://github.com/containers/aardvark-dns/)

This document is not specific to any language; language-specific guidelines are contained in other files.
Rules specific to the Rust language can be found [here](./CONTRIBUTING_RUST.md).

> Rules specific to the Go language can be found [here](./CONTRIBUTING_GO.md).

Reviewer: since podman-py is the only python project we have, I believe including a cross ref to https://github.com/containers/podman-py/blob/main/CONTRIBUTING.md is enough. I wouldn't do CONTRIBUTING_PYTHON.md so maybe here?

Some tests are flaky and will pass on a re-run.

After the reviewers and maintainers take a look, they will either write a comment stating `LGTM` (looks good to me) and approve the PR, in which case you do not need to do any further changes, or they write a comment with review feedback that you should address.
Note that most changes require two reviews so only the second reviewer will actually merge the PR.

If changes were requested, make them locally in your branch and the amend them into the commit from the PR.
You can use `git commit -a --amend` for that.
This will add the current changes to the previous commit.
Please do not push extra commits that say things like "apply code review" or "fix x" where x is a bug introduced in a commit from your PR.
In that case, always squash the change into the right commit to keep the git history clean.
Our projects merge the commits as is and will will not squash them on merge to preserve the full original context.

### Rebasing

When you create a branch to work on a fix or feature, it no longer will be updated with the latest changes from the upstream `main` branch.
In order to keep your branch up to date you should rebase.

> In order to do so add the upstream repo as remote in git, i.e. for containers/netavark use:

Reviewer 2: I think it's an unnecessary change. Maybe it could be like this:
```suggestion
In order to do so add the upstream repo as remote in git, i.e. for containers/<repository-name> use:
```

```

> $ git remote add upstream git@github.com:containers/netavark.git

Reviewer 2: I think it's an unnecessary change. Maybe it could be like this:
```suggestion
$ git remote add upstream git@github.com:containers/<repository-name>.git
```

$ git rebase upstream/main
```

If the PR is open longer you may have to rebase.
You must rebase when there is a merge conflict.
This means the lines that you changed were also changed after you created your branch.
In this case, Git does not know which change is right.
You will need to manually resolve it.
Check [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line) for more information on how to do this.

It is recommended to always rebase on a new push to ensure it is testing against the latest code.

## Find bad changes with git bisect

git bisect is very powerful command in order to quickly find commits that caused a regression.

> For example, assume you updated Netavark and now something that used to work fine is no longer working.

Reviewer 2: ```suggestion
For example, assume you updated Podman and now something that used to work fine is no longer working.
```

Reviewer 3: This comment is still open.
